docs/static-rendering.md

```diff
@@ -0,0 +1,230 @@
+# Static rendering
+
+Client-side apps all suffer the same performance pitfall of not being able to show any content until the JavaScript is downloaded, parsed and run. The user sees a blank white screen until this has all occurred, which is often followed by a loading indicator while data is fetched from the network. To improve perceived performance, sku renders all the static content (everything not requiring an API call) of your app at build time.
+
+## Configuration
+
+The sku config contains route specific options.
+
+```js
+module.exports = {
+  routes: [
+    {
+      route: '/',
+      name: 'home',
+      entry: 'src/Home.js' // Optional: falls back to entry.client
+    },
+    {
+      route: '/details',
+      name: 'details',
+      entry: 'src/Details.js'
+    }
+  ],
+  environments: ['development', 'production'],
+  sites: ['au', 'jobstreet'],
+  target: 'dist' // Optional, this is the default value
+};
+```
+
+Running `sku build` with the above config will create the following output in your target directory.
+
+```
+├── development
+│   ├── au
+│   │   ├── index.html
+│   │   ├── details
+│   │   |   ├── index.html
+│   ├── jobstreet
+│   │   ├── index.html
+│   │   ├── details
+│   │   |   ├── index.html
+├── production
+│   ├── au
+│   │   ├── index.html
+│   │   ├── details
+│   │   |   ├── index.html
+│   ├── jobstreet
+│   │   ├── index.html
+│   │   ├── details
+│   │   |   ├── index.html
+├── [static-asset].{css,js,jpg,etc}
+```
+
+`environments`, `sites` & `routes` are all optional and will not show up in the build output if not specified.
+
+Note: `sku start` will default to the first `environment` and `site` in your config if provided.
+
+## Render entry
+
+After configuring sku, the render entry needs to return the HTML required to create all the above files. The render entry is set via `entry.render` (default is `src/render.js`).
+
+**Example render entry**
+
+```js
+import React from 'react';
+import { renderToString } from 'react-dom/server';
+import App from './App';
+
+export default {
+  renderApp: ({ environment, site, route }) =>
+    renderToString(<App environment={environment} site={site} route={route} />),
+
+  renderDocument: ({ app, bodyTags, headTags }) => `
+    <!DOCTYPE html>
+    <html>
+      <head>
+        <meta charset="UTF-8">
+        <title>My Awesome Project</title>
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        ${headTags}
+      </head>
+      <body>
+        <div id="app">${app}</div>
+        ${bodyTags}
+      </body>
+    </html>
+  `
+};
+```
+
+## renderApp
+
+The `renderApp` function should return a your application as an HTML string (generally using `React.renderToString`). It can also return other values (eg. extracted meta information, css styles, etc). Anything returned will be available in `renderDocument`. You should not render your whole HTML document in `renderApp`, only return the HTML that is generated by React.
```

Comment:
```suggestion
The `renderApp` function should return your application as an HTML string (generally using `React.renderToString`). It can also return other values (e.g. extracted meta information, CSS styles, etc). Anything returned will be available in `renderDocument`. You should not render your whole HTML document in `renderApp`, only return the HTML that is generated by React.
```